lib/Parse/Lexer.cpp

```diff
@@ -1262,14 +1295,26 @@ unsigned Lexer::lexCharacter(const char *&CurPtr, char StopQuote,
       diagnose(CurPtr-1, diag::lex_unterminated_string);
     return ~1U;
   case '\\':  // Escapes.
+    if (!delimiterMatches(DelimiterLength, CurPtr))
+      return '\\';
```

Comment:
I don't think this is right. For `###" \#(some) "###`, this function should diagnose it as invalid escape instead of just returning `\`.